dGame/dBehaviors/SpeedBehavior.cpp

```diff
@@ -7,85 +7,47 @@
 
 
 void SpeedBehavior::Handle(BehaviorContext* context, RakNet::BitStream* bitStream, BehaviorBranchContext branch) {
-	if (m_AffectsCaster) {
-		branch.target = context->caster;
-	}
+	if (m_AffectsCaster) branch.target = context->caster;
 
 	auto* target = EntityManager::Instance()->GetEntity(branch.target);
-
-	if (target == nullptr) {
-		return;
-	}
+	if (!target) return;
 
 	auto* controllablePhysicsComponent = target->GetComponent<ControllablePhysicsComponent>();
-
-	if (controllablePhysicsComponent == nullptr) {
-		return;
-	}
+	if (!controllablePhysicsComponent) return;
 
 	const auto current = controllablePhysicsComponent->GetSpeedMultiplier();
-
 	controllablePhysicsComponent->SetSpeedMultiplier(current + ((m_RunSpeed - 500.0f) / 500.0f));
-
 	EntityManager::Instance()->SerializeEntity(target);
 
 	if (branch.duration > 0.0f) {
 		context->RegisterTimerBehavior(this, branch);
 	} else if (branch.start > 0) {
-		controllablePhysicsComponent->SetIgnoreMultipliers(true);
-
 		context->RegisterEndBehavior(this, branch);
 	}
 }
 
-void SpeedBehavior::Timer(BehaviorContext* context, BehaviorBranchContext branch, LWOOBJID second) {
-	auto* target = EntityManager::Instance()->GetEntity(branch.target);
-
-	if (target == nullptr) {
-		return;
-	}
-
-	auto* controllablePhysicsComponent = target->GetComponent<ControllablePhysicsComponent>();
-
-	if (controllablePhysicsComponent == nullptr) {
-		return;
-	}
-
-	const auto current = controllablePhysicsComponent->GetSpeedMultiplier();
-
-	controllablePhysicsComponent->SetSpeedMultiplier(current - ((m_RunSpeed - 500.0f) / 500.0f));
-
-	EntityManager::Instance()->SerializeEntity(target);
+void SpeedBehavior::Calculate(BehaviorContext* context, RakNet::BitStream* bitStream, BehaviorBranchContext branch) {
+	Handle(context, bitStream, branch);
 }
 
 void SpeedBehavior::End(BehaviorContext* context, BehaviorBranchContext branch, LWOOBJID second) {
 	auto* target = EntityManager::Instance()->GetEntity(branch.target);
-
-	if (target == nullptr) {
-		return;
-	}
+	if (!target) return;
 
 	auto* controllablePhysicsComponent = target->GetComponent<ControllablePhysicsComponent>();
-
-	if (controllablePhysicsComponent == nullptr) {
-		return;
-	}
+	if (!controllablePhysicsComponent) return;
 
 	const auto current = controllablePhysicsComponent->GetSpeedMultiplier();
-
 	controllablePhysicsComponent->SetIgnoreMultipliers(false);
-
 	controllablePhysicsComponent->SetSpeedMultiplier(current - ((m_RunSpeed - 500.0f) / 500.0f));
-
 	EntityManager::Instance()->SerializeEntity(target);
 }
 
+void SpeedBehavior::Timer(BehaviorContext* context, BehaviorBranchContext branch, LWOOBJID second) {
+	End(context, branch, second);
+}
+
 void SpeedBehavior::Load() {
 	m_RunSpeed = GetFloat("run_speed");
-
-	if (m_RunSpeed < 500.0f) {
-		m_RunSpeed = 500.0f;
```

Comment:
Any particular reason this was removed?

Reply:
To allow slower speeds, which had multiple checks to make it not happen, even though it should be possible. this was originally because the wormholer animations would break, but that doesn't happen anymore.
there is one check for slower speeds left in the controllable physics component, this is due the infinite speed glitch that is present, since the infinite can be both positive and negative